support-workers/src/main/scala/com/gu/zuora/ProductSubscriptionBuilders.scala

```diff
@@ -68,53 +74,74 @@ object ProductSubscriptionBuilders {
         contractEffectiveDate = contractEffectiveDate
       )
 
-      val redeemedSub = paymentMethod.fold(
-        _ => subscriptionData.subscription,
-        redemptionData => redemptionData.redeem(subscriptionData.subscription)
+      paymentMethod.fold({ case (_, maybePromoCode) =>
```

Comment:
although fold is more high level, I think a pattern match might be clearer here.